pkg/nad/masterplugin.go

```diff
@@ -174,10 +175,106 @@ func (plugin *MasterBridgePlugin) WithIPAM(ipam *IPAM) *MasterBridgePlugin {
 	if ipam == nil {
 		glog.V(100).Infof("error adding empty ipam to MasterBridgePlugin")
 
-		plugin.errorMsg = "invalid ipam parameter"
+		plugin.errorMsg = invalidIpamParameterMsg
 	}
 
 	plugin.masterPlugin.Ipam = ipam
 
 	return plugin
 }
+
+// MasterVlanPlugin provides struct for MasterPlugin set to vlan in NetworkAttachmentDefinition.
+type MasterVlanPlugin struct {
+	masterPlugin *MasterPlugin
+	errorMsg     string
+}
+
+// NewMasterVlanPlugin creates new instance of MasterVlanPlugin.
+func NewMasterVlanPlugin(name string, vlanID uint16) *MasterVlanPlugin {
+	glog.V(100).Infof(
+		"Initializing new MasterVlanPlugin structure %s, with vlanId %s", vlanID)
+
+	builder := MasterVlanPlugin{
+		masterPlugin: &MasterPlugin{
+			CniVersion: "0.3.1",
+			Name:       name,
+			Type:       "vlan",
+			VlanID:     vlanID,
+		},
+	}
+
+	if vlanID > 4094 {
+		glog.V(100).Infof("error vlan id can not be greater than 4094")
+
+		builder.errorMsg = "MasterVlanPlugin vlanId is greater than 4094"
+	}
+
+	if builder.masterPlugin.Name == "" {
+		glog.V(100).Infof("error MasterVlanPlugin can not be empty")
+
+		builder.errorMsg = "MasterVlanPlugin name is empty"
+	}
+
+	return &builder
+}
+
+// WithIPAM defines IPAM configuration to MasterVlanPlugin. Default is empty.
+func (plugin *MasterVlanPlugin) WithIPAM(ipam *IPAM) *MasterVlanPlugin {
+	glog.V(100).Infof("Adding ipam configuration %v to MasterVlanPlugin", ipam)
```

Comment:
1) IPAM configuration
2) ipam is a struct, so I think better is:
` glog.V(100).Infof("Adding IPAM configuration  to MasterVlanPlugin: %v", ipam)`